components/zcash_address/CHANGELOG.md

```diff
@@ -7,6 +7,30 @@ and this library adheres to Rust's notion of
 
 ## [Unreleased]
 
+### Added
+- `zcash_address::unified`:
+  - `Address::receivers`
+  - `Container::revision`
+  - `DataTypecode`
+  - `Item`
+  - `MetadataItem`
+  - `MetadataTypecode`
+  - `Revision`
+- `zcash_address::convert::TryFromRawAddress::try_from_raw_tex`
+- `zcash_address::convert::TryFromAddress::try_from_tex`
+- `zcash_address::convert::ToAddress::from_tex`
+
+### Changed
+- `zcash_address::unified`:
+  - `Typecode` has changed. Instead of having a variant for each receiver type,
+    it now has two variants, `Typecode::Data` and `Typecode::Metadata`.
+  - `Encoding::try_from_items` now takes an additional `Revision` argument.
+
+### Removed
+- `zcash_address::unified::Container::items` Preference order is only 
+  significant when considering unified address receivers; use
+  `Address::receivers` instead.
```

Comment:
These changes prevent us from introducing TEX addresses in a point release, as they are not backwards-compatible. I would prefer that we separate out the TEX address parts from the ZIP 316 rev1 parts; the former can be deployed as 0.3.2 as intended (#1221), while the latter are required (for this current implementation) to be in 0.4.0.

Reply:
I confirmed locally that the only conflict between 35ea09a16cce24b11c64249020e1ce9f8f557027 and the rest of the PR is in the changelog. Please extract that commit into a separate PR.

Reply:
@str4d does this really matter? It's extra effort to perform the extraction, and I don't see the benefit.

Reply:
I agree with @nuttycom. A breaking update of `zcash_address` is not likely to impose much burden on its current users.

Reply:
I did the extraction locally before making my second comment; it was trivial (with a minor adjustment afterwards because @daira had suggested TEX changes here, that would have been instead made on #1226 had it been opened as a separate PR initially).

> A breaking update of zcash_address is not likely to impose much burden on its current users.

The entire point of the `zcash_address` crate's approach, that TEX addresses are being added to, is to ensure that even if the direct users of `zcash_address` don't perform the breaking change upgrade, _their_ downstream users (i.e. people using Zcash) benefit as a side-effect of the end binary app's `cargo update` process. It's never been about the direct users of `zcash_address`, but about improving the UX for end users: the user learns "hey, this address is unsupported, I should ask APP_NAME to upgrade", not "oh, this address is invalid, I must have done the wrong thing".